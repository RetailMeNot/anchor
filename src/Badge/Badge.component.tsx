// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled from 'styled-components';
// COMPONENTS
// THEME
import { fonts, colors, sizes, variables } from '../theme';

interface BadgeProps {
  count?: string | number;
  className?: string;
  children?: any;
}

export const StyledBadge = styled.div`
  position: relative;
  font-family: ${fonts.fontFamily};
  display: inline-block;

  .badge {
    position: absolute;
    top: -.5rem;
    right: -.25rem;
    background-color: ${colors.secondary.flashPink.base};
    color: ${colors.neutrals.white.base};
    font-weight: 700;
    font-size: 0.75rem;
    padding: 1px ${sizes.padding.xs};
    border-radius: ${variables.borderRadius};
  }
`;

const Badge  = ({ className, children, count, ...props }: BadgeProps) => (
  <StyledBadge className={classNames(className)} {...props}>
    {children}
    <div className="badge">{count}</div>
  </StyledBadge>
);
export default Badge;
