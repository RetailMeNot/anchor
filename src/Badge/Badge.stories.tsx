// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
// VENDOR
import styled from 'styled-components';
// COMPONENTS
import Badge from './Badge.component';
import { Icon } from '../Icon';
// THEME

const StyledStory = styled.div`
  padding: 2rem 5rem;
`;

storiesOf('Components/Badge', module)
  .add('Default', () => (
      <StyledStory>
        <p>Badge 1</p>
        <Badge count={1}>
          <Icon type="cart" />
        </Badge>
      </StyledStory>
  ));
