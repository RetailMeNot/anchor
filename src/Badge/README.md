# Badge

The Badge component displays a dot or a circle with a count. It is normally used to indicate pending items (items in cart, unread messages, etc.)

## Usage

There are three sizes:

-   dot (default)
-   small
-   large

`dot` and `small` are meant to be used with an icon. `large` is meant to be used standalone.

### `dot` with an icon

```jsx
import { Badge } from '@retailmenot/anchor';
import { Cart } from '@retailmenot/anchor';

const BadgeDotExample = props => (
    <Badge
        borderColor="primary.base"
        borderColorHover="primary.light"
        offsetBottom={1.2}
        offsetLeft={0.9}
    >
        <Cart scale="lg" />
    </Badge>
);
```

### `small` with an icon

```jsx
import { Badge } from '@retailmenot/anchor';
import { Cart } from '@retailmenot/anchor';

const BadgeSmallExample = props => (
    <Badge
        borderColor="primary.base"
        borderColorHover="primary.light"
        count={1}
        offsetBottom={1.1}
        offsetLeft={0.6}
        size="small"
        textColor="white"
    >
        <Cart scale="lg" />
    </Badge>
);
```

### `large` standalone

```jsx
import { Badge } from '@retailmenot/anchor';

const BadgeLargeStandaloneExample = props => (
    <Badge
        borderColor="primary.base"
        borderColorHover="primary.light"
        count={1}
        offsetBottom={1.1}
        offsetLeft={0.6}
        size="large"
        standalone={true}
        textColor="white"
    >
        <Cart scale="lg" />
    </Badge>
);
```

## Props

-   **`backgroundColor`**: the background color of the Badge
-   **`borderColor`**: the color of the border of the Badge
-   **`borderColorHover`**: the color of the border of the Badge on hover
-   **`children`**: any elements inside the `<Badge></Badge>` tags. This is what we use to pass `<Icon>` components in.
-   **`className`**: additional className to be added to the Badge element
-   **`count = 0`**: the number to be displayed inside the badge
-   **`isParentHovered`**: boolean that reflects whether the parent element is hovered. This is necessary to change the Badge border color when the the Badge itself is not hovered, but the parent is. See code of stories for an example of how to do this.
-   **`overflowCount = 9`**: the maximum count to show in the Badge. Playing with `count` and `overflowCount` with the Storybook knobs is a good way to test this out.
-   **`offsetBottom`**: the absolute positioning in REM from the bottom. Because of the various sizes of icons and Badges, it seemed fitting to make this configurable.
-   **`offsetLeft`**: the absolute position in REM from the left.
-   **`size = 'dot'`**: the size of the badge. Other options besides `dot` are `small` and `large`.
-   **`standalone = false`**: boolean that represents whether the Badge will be alone.
-   **`textColor = white`**: the color of the text of the `count`.
