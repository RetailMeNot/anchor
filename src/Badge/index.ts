export { Badge } from './Badge.component';
