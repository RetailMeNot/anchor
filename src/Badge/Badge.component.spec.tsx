// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// COMPONENT
import Badge from './Badge.component';
// ENZYME
import { shallow, mount } from 'enzyme';

// TEST SETUP
const subject = <Badge />;
const wrapper = mount(subject);
const component = shallow(subject);

describe('Component: Badge', () => {
    it('should be defined', () => {
        /* tslint:disable */
        console.log(subject);
        console.log(wrapper.debug());
        console.log(component.debug());

        /* tslint:enable */
        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
