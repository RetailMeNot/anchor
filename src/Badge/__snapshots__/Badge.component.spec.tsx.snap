// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Badge should be able to render with an icon 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  height: 0.75rem;
  width: 0.75rem;
}

.c2:hover {
  border: 0.125rem solid #CC3B3B;
}

.c1 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
  color: currentColor;
}

<div
  className="c0 anchor-badge"
>
  <span
    className="c1 anchor-icon cart"
    scale="lg"
  >
    <svg
      height={24}
      viewBox="0 0 16 16"
      width={24}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M6.889 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM11.331 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM4.216 2.681h9.71a.874.874 0 0 1 .853 1.064L13.626 8.93a.874.874 0 0 1-.854.685h-7.49l.187 1.215h7.346a.504.504 0 1 1 0 1.007H5.037a.504.504 0 0 1-.498-.427L3.123 2.207h-1.42a.504.504 0 1 1 0-1.007h1.853c.248 0 .46.181.497.427l.163 1.054zm.913 5.926h7.537l1.093-4.918H4.372l.757 4.918z"
          />
        </g>
      </g>
    </svg>
  </span>
  <div
    className="c2 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should be defined and match its snapshot 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  height: 0.75rem;
  width: 0.75rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should hide the count by default if \`count\` is 0 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 0.4rem;
  display: none;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  font-size: 0.625rem;
  height: 1.25rem;
  min-width: 1.25rem;
  padding: 0 0.25rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="small"
  >
    0
  </div>
</div>
`;

exports[`Component: Badge should render \`count\` if \`count\` is less than \`overflowCount\` 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  font-size: 0.75rem;
  height: 1.75rem;
  min-width: 1.75rem;
  padding: 0 0.5rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="large"
  >
    3
  </div>
</div>
`;

exports[`Component: Badge should render \`overflowCount\`+ if \`count\` is more than \`overflowCount\` 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  font-size: 0.75rem;
  height: 1.75rem;
  min-width: 1.75rem;
  padding: 0 0.5rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="large"
  >
    6+
  </div>
</div>
`;

exports[`Component: Badge should render a count with the border color of \`borderColorHover\` if \`isParentHovered\` is true. 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid palevioletred;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  height: 0.75rem;
  width: 0.75rem;
}

.c1:hover {
  border: 0.125rem solid palevioletred;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should render a dot as the default \`size\` 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  height: 0.75rem;
  width: 0.75rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should render a double-digit count in a small badge with fully rounded border-radius 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: 1.1rem;
  left: 0.6rem;
  font-size: 0.625rem;
  height: 1.25rem;
  min-width: 1.25rem;
  padding: 0 0.25rem;
}

.c2:hover {
  border: 0.125rem solid #CC3B3B;
}

.c1 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
  color: currentColor;
}

<div
  className="c0 anchor-badge"
>
  <span
    className="c1 anchor-icon cart"
    scale="lg"
  >
    <svg
      height={24}
      viewBox="0 0 16 16"
      width={24}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M6.889 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM11.331 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM4.216 2.681h9.71a.874.874 0 0 1 .853 1.064L13.626 8.93a.874.874 0 0 1-.854.685h-7.49l.187 1.215h7.346a.504.504 0 1 1 0 1.007H5.037a.504.504 0 0 1-.498-.427L3.123 2.207h-1.42a.504.504 0 1 1 0-1.007h1.853c.248 0 .46.181.497.427l.163 1.054zm.913 5.926h7.537l1.093-4.918H4.372l.757 4.918z"
          />
        </g>
      </g>
    </svg>
  </span>
  <div
    className="c2 anchor-badge-count"
    size="small"
  >
    9+
  </div>
</div>
`;

exports[`Component: Badge should render a single-digit count in a large badge with fully rounded border-radius 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: 1.1rem;
  left: 0.6rem;
  font-size: 0.75rem;
  height: 1.75rem;
  min-width: 1.75rem;
  padding: 0 0.5rem;
}

.c2:hover {
  border: 0.125rem solid #CC3B3B;
}

.c1 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
  color: currentColor;
}

<div
  className="c0 anchor-badge"
>
  <span
    className="c1 anchor-icon cart"
    scale="lg"
  >
    <svg
      height={24}
      viewBox="0 0 16 16"
      width={24}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M6.889 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM11.331 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM4.216 2.681h9.71a.874.874 0 0 1 .853 1.064L13.626 8.93a.874.874 0 0 1-.854.685h-7.49l.187 1.215h7.346a.504.504 0 1 1 0 1.007H5.037a.504.504 0 0 1-.498-.427L3.123 2.207h-1.42a.504.504 0 1 1 0-1.007h1.853c.248 0 .46.181.497.427l.163 1.054zm.913 5.926h7.537l1.093-4.918H4.372l.757 4.918z"
          />
        </g>
      </g>
    </svg>
  </span>
  <div
    className="c2 anchor-badge-count"
    size="large"
  >
    2
  </div>
</div>
`;

exports[`Component: Badge should render a single-digit count in a small badge with only partially rounded border-radius 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: 1.1rem;
  left: 0.6rem;
  font-size: 0.625rem;
  height: 1.25rem;
  min-width: 1.25rem;
  padding: 0 0.25rem;
}

.c2:hover {
  border: 0.125rem solid #CC3B3B;
}

.c1 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
  color: currentColor;
}

<div
  className="c0 anchor-badge"
>
  <span
    className="c1 anchor-icon cart"
    scale="lg"
  >
    <svg
      height={24}
      viewBox="0 0 16 16"
      width={24}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M6.889 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM11.331 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM4.216 2.681h9.71a.874.874 0 0 1 .853 1.064L13.626 8.93a.874.874 0 0 1-.854.685h-7.49l.187 1.215h7.346a.504.504 0 1 1 0 1.007H5.037a.504.504 0 0 1-.498-.427L3.123 2.207h-1.42a.504.504 0 1 1 0-1.007h1.853c.248 0 .46.181.497.427l.163 1.054zm.913 5.926h7.537l1.093-4.918H4.372l.757 4.918z"
          />
        </g>
      </g>
    </svg>
  </span>
  <div
    className="c2 anchor-badge-count"
    size="small"
  >
    5
  </div>
</div>
`;

exports[`Component: Badge should render the count with absolute positioning if \`standalone\` is false 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: 1.2rem;
  left: 0.9rem;
  height: 0.75rem;
  width: 0.75rem;
}

.c2:hover {
  border: 0.125rem solid #CC3B3B;
}

.c1 {
  display: inline-block;
  height: 1.5rem;
  width: 1.5rem;
  line-height: 0;
  color: currentColor;
}

<div
  className="c0 anchor-badge"
>
  <span
    className="c1 anchor-icon cart"
    scale="lg"
  >
    <svg
      height={24}
      viewBox="0 0 16 16"
      width={24}
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h16v16H0z"
        />
        <g
          fill="currentColor"
          fillRule="nonzero"
        >
          <path
            d="M6.889 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM11.331 14.667a1.111 1.111 0 1 1 0-2.223 1.111 1.111 0 0 1 0 2.223zM4.216 2.681h9.71a.874.874 0 0 1 .853 1.064L13.626 8.93a.874.874 0 0 1-.854.685h-7.49l.187 1.215h7.346a.504.504 0 1 1 0 1.007H5.037a.504.504 0 0 1-.498-.427L3.123 2.207h-1.42a.504.504 0 1 1 0-1.007h1.853c.248 0 .46.181.497.427l.163 1.054zm.913 5.926h7.537l1.093-4.918H4.372l.757 4.918z"
          />
        </g>
      </g>
    </svg>
  </span>
  <div
    className="c2 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should render the count without absolute positioning if \`standalone\` is true 1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.75rem;
  width: 0.75rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="dot"
  >
    
  </div>
</div>
`;

exports[`Component: Badge should show the count if \`showZero\` is true, even if \`count\` is 0  1`] = `
.c0 {
  position: relative;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  display: inline-block;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #CC3B3B;
  color: white;
  border: 0.125rem solid #CC3B3B;
  box-sizing: border-box;
  font-weight: 700;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  border-radius: 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  bottom: undefinedrem;
  left: undefinedrem;
  font-size: 0.625rem;
  height: 1.25rem;
  min-width: 1.25rem;
  padding: 0 0.25rem;
}

.c1:hover {
  border: 0.125rem solid #CC3B3B;
}

<div
  className="c0 anchor-badge"
>
  <div
    className="c1 anchor-badge-count"
    size="small"
  >
    0
  </div>
</div>
`;
